src/libraries/System.Private.CoreLib/src/System/Text/DecoderFallback.cs

```diff
@@ -11,15 +11,12 @@ namespace System.Text
 {
     public abstract class DecoderFallback
     {
-        private static DecoderFallback? s_replacementFallback; // Default fallback, uses no best fit & "?"
-        private static DecoderFallback? s_exceptionFallback;
+        // Default fallback, uses no best fit & "?"
+        private static readonly DecoderReplacementFallback s_replacementFallback = new DecoderReplacementFallback();
+        public static DecoderFallback ReplacementFallback => s_replacementFallback;
 
-        public static DecoderFallback ReplacementFallback =>
-            s_replacementFallback ?? Interlocked.CompareExchange(ref s_replacementFallback, new DecoderReplacementFallback(), null) ?? s_replacementFallback;
-
-
-        public static DecoderFallback ExceptionFallback =>
-            s_exceptionFallback ?? Interlocked.CompareExchange<DecoderFallback?>(ref s_exceptionFallback, new DecoderExceptionFallback(), null) ?? s_exceptionFallback;
+        private static readonly DecoderExceptionFallback s_exceptionFallback = new DecoderExceptionFallback();
+        public static DecoderFallback ExceptionFallback => s_exceptionFallback;
```

Comment:
Do typical apps need both `DecoderExceptionFallback` and `DecoderExceptionFallback`?  I guess it was split so that IL linker can remove it when it is not used.

cc @marek-safar

Reply:
It's referenced by the `UTF8Encoding` ctor.

https://github.com/dotnet/runtime/blob/ac78451b9354f150b892bf1e4ffc1a86e2db848a/src/libraries/System.Private.CoreLib/src/System/Text/UTF8Encoding.cs#L85-L108

Reply:
The way to do this in IL linker way is to define the singleton inside the class that the singleton belongs to. Ie, this would become: `public static DecoderFallback ReplacementFallback => DecoderReplacementFallback.s_default;`.

Reply:
Jan is correct that the previous implementation is size friendly. I don't have strong preference between CAS and single-purpose static class, although CAS implementation might be more size friendly.